internal/pkg/config/config.go

```diff
@@ -29,15 +29,40 @@ import (
 	"github.com/google/go-flow-levee/internal/pkg/config/regexp"
 )
 
-// FlagSet should be used by analyzers to reuse -config flag.
-var FlagSet flag.FlagSet
-var configFile string
+var (
+	// FlagSet should be used by analyzers to reuse -config flag.
+	FlagSet                    flag.FlagSet
+	configFile                 string
+	validFuncMatcherFields     map[string]bool
+	validSourceMatcherFields   map[string]bool
+	validFieldTagMatcherFields map[string]bool
+)
 
 func init() {
 	FlagSet.StringVar(&configFile, "config", "config.yaml", "path to analysis configuration file")
+	validFuncMatcherFields = map[string]bool{
```

Comment:
Have you considered writing each field once in title case and using e.g. `strings.Title` when doing the comparisons?

Reply:
maybe we should just compare case-insensitively?

Reply:
Right. I don't think we really care about the casing, so maybe casefolding is the way to go.
@PurelyApplied, any thoughts on this?

Reply:
I don't really have an opinion on it one way or t'other.  We wound up with case insensitivity as a side-benefit of using the `k8s` YAML package rather than the `go-yaml` package.  But that choice was driven by not wanting to have to tag every single field with

```golang
type Foo struct{
  FieldName string `yaml: "FieldName"`
}
```

The `go-yaml` unmarshaller gets very upset at in-name capitalization that you have to go manually correct.

-----

All this to say, case folding sounds fine to me.

Reply:
Cool. Added casefolding to comparison.